### DISCUSSION TITLE
CURLOPT_ACCEPT_ENCODING - how to tell if response automatically decompressed?

### DISCUSSION THREAD

**Discussion by bkdotcom** [curl/curl#15738] [General]
if I set `CURLOPT_ACCEPT_ENCODING` = "br"   (or some other encoding)

how can I tell if curl automaticaly decompressed the response?

curl leaves the original `Content-Encoding: br`   header in the response when it decompresses the response

(I argue it should rename the header to `x-content-encoding` if it was able to uncompress)

**Comment by icing** [comment]
I'd say the header should be reported *exactly* as they were received by the server. But I agree that there should be an indication for the libcurl application that decompression has taken place.
